design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,169 @@
+
+This example `object-store-realm.yaml`, names a realm `my-realm`.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+
+1. At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will start creating the resource to start the object store realm.
+
+2. After these steps the admin should start up a master [object-store-zonegroup](/design/ceph/object-store-zonegroup.md), a master [object-store-zone](/design/ceph/object-store-zone.md), and a [object-store](/design/ceph/object-store.md) in the object-store-zone resource.
+
+3. Once all of the resources from step #2 are created, the operator will create a realm on the Rook Ceph cluster and the resource will be running.
+
+#### Notes
+
+1. If a realm has already been created or pulled with the same name on the cluster, creation of this resource will fail.
+
+### Pulling the Realm from another Ceph Cluster
+
+#### Config
+
+When the storage admin is ready to sync data from another Ceph cluster with multisite set up, the first step is to pull the realm with the data from that cluster.
+
+To do this, the storage admin needs to create an object-store-realm with the same name as the realm being pulled from. This resource must have the `pullEndpoint` section configured. 
+
+The endpoint in the `pullEndpoint` section is an endpoint of an object-store in the master-zone. This endpoint must be resolvable.
+
+This example `object-store-realm.yaml`, pulls `my-realm` from a Ceph cluster with endpoint http://my-realm-zone-b-objectstore-1:80 in its master zone.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph-2
+spec:
+  pullEndpoint: http://my-realm-zone-b-objectstore-1:80
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+1. At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will start creating the resource to start the object store realm.
+
+2. After these steps the admin should start up a [object-store-zonegroup](/design/ceph/object-store-zonegroup.md). This zonegroup should be the same zonegroup of the zone the endpoint for in the pullRealm section is in.
+
+3. After the object-store-zonegroup resource is started, a non-master [object-store-zone](/design/ceph/object-store-zone.md), and a [object-store](/design/ceph/object-store.md) in the object-store-zone resource need to be started up.
+
+3. Once all of the resources steps #2 & #3 are created, the operator will pull the realm on the rook Ceph cluster and the resource will be running.
+
```

Comment:
If we use "multisite", we will use the pullEndpint to retrieve the realm data.
Could it be possible to avoid to configure the zone group  (because it will be the same that in the pulled realm)?

Reply:
We discussed this with @yuvalif when we were working on the design. We wanted the user to have the same realm and zonegroup resources on a pulling cluster as the primary cluster. I'd be happy to discuss this further.

Also the object-store-zone needs a zonegroup in it's config. If the object-store-zonegroup CR hasn't been configured yet, the object-store-zone won't get created.

Reply:
> Yes the user should have the same realm and zonegroup across both clusters, but the user should only need to define it once in the master cluster. After doing a pull (see above comment about `CephRealmPull`, the operator should create both the realm and zonegroup automatically.
> 
agree. the resource should be there, but it should be created automatically upon the "pull". only explicit definition should be of the "zone" and "object store".

> I suppose this is the important aspect in my mind. I see the operator as a _bidirectional_ operator that listens for changes in both ceph and k8s. In that way changes on one k8s cluster will propagate to the other in the same way that periods propagate in ceph.